python-package/lightgbm/__init__.py

```diff
@@ -32,5 +32,5 @@
            'train', 'cv',
            'LGBMModel', 'LGBMRegressor', 'LGBMClassifier', 'LGBMRanker',
            'DaskLGBMRegressor', 'DaskLGBMClassifier', 'DaskLGBMRanker',
-           'log_evaluation', 'record_evaluation', 'reset_parameter', 'early_stopping',
+           'log_evaluation', 'record_evaluation', 'reset_parameter', 'early_stopping', 'EarlyStopException',
```

Comment:
This allows for imports like

```python
from lightgbm import EarlyStopException
```

or

```python
import lightgbm as lgb

def _early_stop(env):
    raise lgb.EarlyStopException(
        best_iteration=6,
        best_score=[("some_validation_set", "some_metric", 0.708, True)]
     )
```

Every other part of the public API can be imported like this, and I think such imports should be encouraged over those that require knowing the internal module structure like `from lightgbm.callback import EarlyStopException`